src/linux/device_info.cpp

```diff
@@ -0,0 +1,127 @@
+#include "device_info.h"
+#include <QFile>
+#include <QTextStream>
+#include <QJsonObject>
+#include <QDebug>
+
+device_info::device_info()
+{
+    _is_raspberry_pi = false;
+    _revision = "";
+    _device_code = "";
+    _tags = {};
+    _tags_set = false;
+    _hardware_name = "";
+    
+}
+
+bool device_info::determineHardware()
+{
+    // Attempt to open /proc/cpuinfo
+    QFile f("/proc/cpuinfo");
+    if (f.open(QFile::ReadOnly | QFile::Text)) {
+        QTextStream s(&f);
+        while (!s.atEnd()) {
+            QString line = s.readLine();
+            if (line.startsWith("Revision")) {
+                _revision = line.section(':', 1).trimmed();
+                bool ok;
+                uint32_t value_int = _revision.toUInt(&ok, 16);
+
+                // Device type is 8 bits, stored in bits 4-11
+                const uint32_t MASK = 0xFF << 4;
+                uint32_t isolated_bits = value_int & MASK;
+                uint32_t extracted_value = isolated_bits >> 4;

```

Comment:
I guess this is assuming that the device is using a new-style revision code. Should we also support old-style revision codes, or do they not matter as those devices probably won't be able to run the Embedded Imager anyway?
https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-revision-codes
At the very least, you should probably check that the "New flag" (bit 23) is set, before trying to read bits 4-11 ?